### PR TITLE
Improve docs for the action-mail box incineration

### PR DESCRIPTION
**Pull request by yoshixi**
### Motivation / Background

This Pull Request has been created because the action-mailbox incineration feature's description is a little vague.

### Detail

This Pull Request changes the description of how the incineration is invoked to be more clearly.

### Checklist

Before submitting the PR make sure the following are checked:

* [x] This Pull Request is related to one change. Changes that are unrelated should be opened in separate PRs.
* [x] Commit message has a detailed description of what changed and why. If this PR fixes a related issue include it in the commit message. Ex: `[Fix #issue-number]`
* [x] Tests are added or updated if you fix a bug or add a feature.
* [x] CHANGELOG files are updated for the changed libraries if there is a behavior change or additional feature. Minor bug fixes and documentation changes should not be included.

### CODE REVIEW

File: guides/source/action_mailbox_basics.md [hunk 1]
```diff
@@ -324,9 +324,10 @@ end
 
 ## Incineration of InboundEmails
 
-By default, an InboundEmail that has been successfully processed will be
-incinerated after 30 days. This ensures you're not holding on to people's data
-willy-nilly after they may have canceled their accounts or deleted their
+By default, an InboundEmail that has been processed will be
+incinerated after 30 days, regardless of whether the procession succeeds or fails.
```
Comment (yoshixi) [inline guides/source/action_mailbox_basics.md:328]: I found that incineration happens regardless of whether the process is successful or not. So, I tried to describe it more clearly.

https://github.com/rails/rails/blob/6f6c211f4781e7bda63e5f47089f9d2612dc0a52/actionmailbox/app/models/action_mailbox/inbound_email/incineratable.rb#L10C5-L10C24

File: guides/source/action_mailbox_basics.md [hunk 2]
```diff
@@ -324,9 +324,10 @@ end
 
 ## Incineration of InboundEmails
 
-By default, an InboundEmail that has been successfully processed will be
-incinerated after 30 days. This ensures you're not holding on to people's data
-willy-nilly after they may have canceled their accounts or deleted their
+By default, an InboundEmail that has been processed will be
```
Comment (vipulnsward) [inline guides/source/action_mailbox_basics.md:327]: The "processing" here means `processed?` status- https://github.com/rails/rails/blob/6f6c211f4781e7bda63e5f47089f9d2612dc0a52/actionmailbox/app/models/action_mailbox/inbound_email.rb#L44-L44

Think it is correct as is.

Comment (yoshixi) [reply guides/source/action_mailbox_basics.md:327]: The `processed?` included the fails status. It was a little bit confusing to me. Even if the job fails with exceptional error, incineration happens. I thought this did not match the current doc.

Comment (yoshixi) [reply guides/source/action_mailbox_basics.md:327]: But I can get what you mean, so this PR should be improved.

Comment (yoshixi) [reply guides/source/action_mailbox_basics.md:327]: I fixed it.
https://github.com/rails/rails/pull/49743/commits/9153b8f9c2f812d1d42466a65de74f3e88d3026d

File: guides/source/action_mailbox_basics.md [hunk 3]
```diff
@@ -324,9 +324,11 @@ end
...
+incinerated after 30 days. The InboundEmail is considered as processed
```
Comment (vipulnsward) [inline guides/source/action_mailbox_basics.md:328]: > The InboundEmail is considered as processed whether the job processing fails or succeeds.

The InboundEmail is considered as processed when its status changes to delivered, failed or bounced.

**Comment by vipulnsward** [comment]
Hi @yoshixi can you add above change and squash your commits, thanks!

**Comment by yoshixi** [comment]
@vipulnsward I applied. Thank you for taking a look at it!

**Comment by zzak** [comment]
Why not link to the `processed?` API docs, so the reader can see the actual definition:
https://edgeapi.rubyonrails.org/classes/ActionMailbox/InboundEmail.html#method-i-processed-3F

This class may also be updated:
https://edgeapi.rubyonrails.org/classes/ActionMailbox/InboundEmail/Incineratable.html

**Comment by yoshixi** [comment]
@vipulnsward 
I linked this document. I thought this document was appropriate to attach to the section I modified.

https://edgeapi.rubyonrails.org/classes/ActionMailbox/InboundEmail.html

**Review by zzak — APPROVED** [review]

**Pull request merged**